plutus-use-cases/src/Plutus/Contracts/Auction.hs

```diff
@@ -71,6 +77,24 @@ data AuctionState
     deriving stock (Generic, Haskell.Show, Haskell.Eq)
     deriving anyclass (ToJSON, FromJSON)
 
+-- | Observable state of the auction app
+data AuctionOutput =
+    AuctionOutput
+        { auctionState       :: Last AuctionState
+        , auctionThreadToken :: Last Currency
```

Comment:
Hm, this is a little funny since you probably shouldn't change this. I guess morally this ought to be one bit of reader input and one bit of state. But at least perhaps this one should be `First Currency`!